src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-logic.cpp

```diff
@@ -0,0 +1,1048 @@
+        handler.AddStatus(commandPath, Status::ConstraintError);
+    });
+
+    std::vector<uint16_t> connectionIDList;
+
+    if (connectionID.IsNull())
+    {
+        for (auto & transportConnection : mCurrentConnections)
+        {
+            if (transportConnection.fabricIndex == handler.GetAccessingFabricIndex())
+            {
+                connectionIDList.push_back(transportConnection.connectionID);
+            }
+        }
+    }
+    else
+    {
+        FabricIndex fabricIndex = handler.GetAccessingFabricIndex();
+        TransportConfigurationStorage * transportConfiguration =
+            FindStreamTransportConnectionWithinFabric(connectionID.Value(), fabricIndex);
+        if (transportConfiguration == nullptr)
+        {
+            ChipLogError(Zcl, "HandleSetTransportStatus[ep=%d]: ConnectionID Not Found.", mEndpointId);
+            handler.AddStatus(commandPath, Status::NotFound);
+            return std::nullopt;
+        }
+        connectionIDList.push_back(connectionID.Value());
+    }
+    // Call the delegate
+    status = mDelegate->SetTransportStatus(connectionIDList, transportStatus);
+    if (status == Status::Success)
+    {
+        for (auto & connID : connectionIDList)
+        {
+            for (auto & transportConnection : mCurrentConnections)
+            {
+                if (transportConnection.connectionID == connID)
+                {
+                    transportConnection.transportStatus = transportStatus;
+                }
+            }
+        }
+    }
+    handler.AddStatus(commandPath, status);
+
+    return std::nullopt;
+}
+
+std::optional<DataModel::ActionReturnStatus> PushAvStreamTransportServerLogic::HandleManuallyTriggerTransport(
+    CommandHandler & handler, const ConcreteCommandPath & commandPath,
+    const Commands::ManuallyTriggerTransport::DecodableType & commandData)
+{
+    if (IsNullDelegateWithLogging(commandPath.mEndpointId))
+    {
+        return Status::UnsupportedCommand;
+    }
+
+    Status status           = Status::Success;
+    uint16_t connectionID   = commandData.connectionID;
+    auto & activationReason = commandData.activationReason;
+
+    VerifyOrDo(activationReason != TriggerActivationReasonEnum::kUnknownEnumValue, {
+        ChipLogError(Zcl, "HandleManuallyTriggerTransport[ep=%d]: Invalid Activation Reason ", mEndpointId);
+        handler.AddStatus(commandPath, Status::ConstraintError);
+    });
+
+    Optional<Structs::TransportMotionTriggerTimeControlStruct::DecodableType> timeControl = commandData.timeControl;
+
+    if (timeControl.HasValue())
+    {
+        VerifyOrDo(timeControl.Value().initialDuration >= 1, {
+            ChipLogError(Zcl, "HandleManuallyTriggerTransport[ep=%d]: Motion Time Control (InitialDuration) Constraint Error",
+                         mEndpointId);
+            handler.AddStatus(commandPath, Status::ConstraintError);
```

Comment:
And return here....  Please check all of these.  I'm going to stop commenting on them; just check them all.